test/test_auth.py

```diff
@@ -475,3 +475,37 @@ def test_auth_dbname_works_fine(
         # The client connects to postgres DB that matches with autodb
         # pgbouncer must use postgres_authdb1, which is defined in [pgbouncer] section
         bouncer.test(user="stats", password="stats", dbname="postgres")
+
+
+@pytest.mark.skipif("WINDOWS", reason="Windows does not have SIGHUP")
+def test_client_hba_cert(bouncer, cert_dir):
```

Comment:
Some additional tests that should be added:
- [ ] Rejects connection with the wrong CN in the cert (e.g. try connecting with localhost cert)
- [ ] Rejects connection when CN matches, but mapname doesn't match.
- [ ] "all" allows connection with any user